packages/smooth_app/lib/pages/product_page.dart

```diff
@@ -83,16 +86,30 @@ class _ProductPageState extends State<ProductPage> {
         ),
       ),
     );
-    if (mainVariables.isNotEmpty) {
+    final Match match =
+        Match(widget.product, userPreferences, userPreferencesModel);
+    for (final String attributeTag in mainVariables) {
       listItems.add(
         AttributeListExpandable(
           product: widget.product,
           iconWidth: iconWidth,
-          attributeTags: mainVariables,
-          title: 'MY PREFERENCES',
+          attributeTags: <String>[attributeTag],
+          collapsible: false,
+          background: PersonalizedRankingPage.getBackgroundColor(
```

Comment:
For the background of each attribute, we use 5 colors based on the score of each attribute.
Here is the corresponding code for the web site: https://github.com/openfoodfacts/openfoodfacts-server/blob/master/html/js/product-search.js#L253

```````
function display_product_summary(target, product) {
	
	var user_product_preferences = get_user_product_preferences();
	
	match_product_to_preferences(product, user_product_preferences);
	
	var attributes_html = '';
		
	$.each(product.match_attributes.mandatory.concat(product.match_attributes.very_important, product.match_attributes.important), function (key, attribute) {
		
		// vary the color from green to red
		var color = "#eee";
		
		if (attribute.status == "known") {
			if (attribute.match <= 20) {
				color = "hsl(0, 100%, 90%)";
			}
			else if (attribute.match <= 40) {
				color = "hsl(30, 100%, 90%)";
			}
			else if (attribute.match <= 60) {
				color = "hsl(60, 100%, 90%)";
			}
			else if (attribute.match <= 80) {
				color = "hsl(90, 100%, 90%)";
			}
			else {
				color = "hsl(120, 100%, 90%)";
			}
		}

		attributes_html += '<li>'
		+ '<div style="border-radius:12px;background-color:' + color + ';padding:1rem;min-height:96px;">'
		+ '<img src="' + attribute.icon_url + '" style="height:72px;float:right;margin-left:0.5rem;">'
		+ '<h4>' + attribute.title + '</h4>';
		
		if (attribute.description_short) {
			attributes_html += '<span>' + attribute.description_short + '</span>';
		}
		
		if (attribute.missing) {
			attributes_html += "<p class='attribute_missing'>" + attribute.missing + "</p>";
		}		

		attributes_html += '</div></li>';
	});
	
	$( target ).html('<ul id="attributes_grid" class="small-block-grid-1 medium-block-grid-2 large-block-grid-3">' + attributes_html + '</ul>');
		}